timeline/plugins.py

```diff
@@ -70,6 +77,58 @@ def get_statistics(self):
             }
         }
 
+    def check_permission(self, user, event):
+        """Check if user has permission to view event"""
+        if event.classified:
```

Comment:
I was able to make the site crash with the timeline search feature, and it seems `check_permission()` is the culprit.

This assumes every event has an associated project, which they of course don't as we can also have site-wide events. So if the search results contain a site-wide event, this will crash with the following:

 ```
AttributeError at /project/search/results/
'NoneType' object has no attribute 'public_guest_access'
```

You should also check for non-project event perms. Please also update tests to include site-wide events to ensure this works.